On Alpine __gnu_linux__ doesn't seem to be defined, but __linux__ is.

diff -upr Ardour-6.5.0.orig/libs/vst3/pluginterfaces/base/fplatform.h Ardour-6.5.0/libs/vst3/pluginterfaces/base/fplatform.h
--- Ardour-6.5.0.orig/libs/vst3/pluginterfaces/base/fplatform.h	2020-12-21 12:58:53.793104277 +0100
+++ Ardour-6.5.0/libs/vst3/pluginterfaces/base/fplatform.h	2020-12-21 13:00:52.453542639 +0100
@@ -86,7 +86,7 @@
 //-----------------------------------------------------------------------------
 // LINUX
 //-----------------------------------------------------------------------------
-#elif __gnu_linux__
+#elif __linux__
 	#define SMTG_OS_LINUX	1
 	#define SMTG_OS_MACOS	0
 	#define SMTG_OS_WINDOWS	0
